pyiceberg/catalog/__init__.py

```diff
@@ -94,6 +94,7 @@
 URI = "uri"
 LOCATION = "location"
 EXTERNAL_TABLE = "EXTERNAL_TABLE"
+MAX_RETRIES = 10
```

Comment:
nit, use the `<>_DEFAULT` convention

https://github.com/apache/iceberg-python/blob/cd21f9706cd4cdc7ee9427b603b3949b005ecda7/pyiceberg/catalog/glue.py#L374

Also not sure if this property belongs in the `__init__` file, unless we're saying that this is the default for all catalogs